docs/sections/components/advanced/argilla.md

```diff
@@ -0,0 +1,118 @@
+
+- **Questions**: `quality` will be the only question for the annotators to answer i.e. to annotate, and it will be an `argilla.LabelQuestion` referring to the quality of the provided generation for the given instruction, and can be annotated with either 👎 (bad) or 👍 (good).
+
+!!! NOTE
+    The [`TextGenerationToArgilla`][distilabel.steps.TextGenerationToArgilla] step will only work as is if the [`Pipeline`][distilabel.pipeline.Pipeline] contains one or multiple [`TextGeneration`][distilabel.steps.tasks.TextGeneration] steps, or if the columns `instruction` and `generation` are available within the batch data. Otherwise, the variable `input_mappings` will need to be set so that either both or one of `instruction` and `generation` are mapped to one of the existing columns in the batch data.
+
+```python
+from distilabel.llms import OpenAILLM
+from distilabel.steps import LoadDataFromDicts, TextGenerationToArgilla
+from distilabel.steps.tasks import TextGeneration
+
+
+with Pipeline(name="my-pipeline") as pipeline:
+    load_dataset = LoadDataFromDicts(
+        name="load_dataset",
+        data=[
+            {
+                "instruction": "Write a short story about a dragon that saves a princess from a tower.",
+            },
+        ],
+    )
+
+    text_generation = TextGeneration(
+        name="text_generation",
+        llm=OpenAILLM(model="gpt-4"),
+    )
+
+    to_argilla = TextGenerationToArgilla(
+        dataset_name="my-dataset",
+        dataset_workspace="admin",
+        api_url="<ARGILLA_API_URL>",
+        api_key="<ARGILLA_API_KEY>",
+    )
+
+    load_dataset >> text_generation >> to_argilla
+
+pipeline.run()
+```
+
+![Text Generation to Argilla](../../../assets/images/sections/learn/steps/argilla/text_generation.png)
+
+### Preference
+
+For preference scenarios, i.e. when the [`Pipeline`][distilabel.pipeline.Pipeline] contains multiple [`TextGeneration`][distilabel.steps.tasks.TextGeneration] steps, we have designed the task [`PreferenceToArgilla`][distilabel.steps.PreferenceToArgilla], which will seamlessly push the generated data to Argilla, and allow the annotator to review the records.
+
+The dataset will be pushed with the following configuration:
+
+- **Fields**: `instruction` and `generations`, both being fields of type `argilla.TextField`, plus the automatically generated `id` for the given `instruction` to be able to search with it. The field `instruction` must always be a string, while the field `generations` must be a list of strings, containing the generated texts for the given `instruction` so that at least there are two generations to compare. Other than that, the number of `generation` fields within each record in Argilla will be defined by the value of the variable `num_generations` to be provided in the [`PreferenceToArgilla`][distilabel.steps.PreferenceToArgilla] step.
```

Comment:
```suggestion
- **Fields**: `instruction` and `generations`, both being fields of type `argilla.TextField`, plus the automatically generated `id` for the given `instruction` to be able to search for other records with the same `instruction` in the dataset. The field `instruction` must always be a string, while the field `generations` must be a list of strings, containing the generated texts for the given `instruction` so that at least there are two generations to compare. Other than that, the number of `generation` fields within each record in Argilla will be defined by the value of the variable `num_generations` to be provided in the [`PreferenceToArgilla`][distilabel.steps.PreferenceToArgilla] step.
```